### DISCUSSION TITLE
About dev only FOUC

### DISCUSSION THREAD

**Discussion by ojj1123** [wakujs/waku#404] [Ideas]
_Related #344_ 

I was investigating dev only FOUC(flash of unstyled contents) issue. I think Waku need to have the logic like this:

> Whenever starting the dev server, running HMR or refreshing, it bundle the css assets and inject the style or link tag in HTML.

This seems like obvious but Waku implements this feature just in PRD: [it build server bundle and then client bundle in PRD](https://github.com/dai-shi/waku/blob/dba04b7b1a55ecc663a802273876730b42b6ffcc/packages/waku/src/lib/builder/build.ts#L540-L560)
I have some questions: **Why did you do that? Do you have any background issue?**

[Solid-start](https://github.com/solidjs/solid-start) team [solves FOUC issue for SSR in Vite dev: walking the module graph based on requested routes, collect relevant styles, inline in a style tag.](https://x.com/markdalgleish/status/1719879166890484088?s=20)

So I was wondering if we could also build client bundle and then inject style or link tag in DEV.
If we can do it, Could you tell me what I need to pay attention to in DEV differently from PRD, or some reference you know?

**Comment by dai-shi** [comment]
Yes, our background is Waku doesn't do anything special for injecting styles and let React hoistable elements (codenamed React Float) do it. We want to follow React features and make Waku as thin as possible.
So, probably instead of fixing FOUC directly, if we fix #113, FOUC will be gone.

**Comment by Aslemammad** [comment]
I guess this is fixed with #427?

**Reply by dai-shi** [reply]
yes, and #439. thanks!
